File: leetcode/easy/javascript/Longest Common Prefix.js
var longestCommonPrefix = function(strs) {
    if(!strs.length) return //strsの配列空なら終了

    return strs.reduce((prev,next)=>{   //reduceメソッドで配列要素を順番に取り出す
        let get = 0

        while(prev[get]&&next[get]&&prev[get]===next[get]) {　//取り出された要素に比較する値が存在しているかつ比較して値が一致している間くりかえす
            get+=1
        }
        return prev.slice(0,get)　//繰り返された数だけ先頭から値を切り抜く
    })
};

File: leetcode/easy/python/Longest Common Prefix.py
class Solution:
    from typing import List
    def longestCommonPrefix(self, strs: List[str]) -> str:
        pre = strs[0] #確認用文字列として配列最初の値を代入

        for i in strs: #引数配列値の内容でループ
            while not i.startswith(pre): #確認用文字列とループ文字列の値が一致していない間
                pre = pre[:-1] #文字を一つ減らして再度代入
        return pre